packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -960,7 +960,16 @@ mixin WidgetInspectorService {
   /// Structured errors provide semantic information that can be used by IDEs
   /// to enhance the display of errors with rich formatting.
   bool isStructuredErrorsEnabled() {
-    return const bool.fromEnvironment('flutter.inspector.structuredErrors');
+    const String extension = 'flutter.inspector.structuredErrors';
+    bool enabled = false;
+    assert(() {
+      enabled = const bool.fromEnvironment(extension, defaultValue: true);
+      return true;
+    }());
+    if (enabled) {
```

Comment:
These 4 lines should be replaced by`return enabled;`

Reply:
if `const bool.fromEnvironment('flutter.inspector.structuredErrors')` is non-null though, we want to return it. If we don't return `const bool.fromEnvironment(extension);` at the end, we will automatically return false in profile mode, even though `bool.fromEnvironment(extension)` may have been true.

Reply:
This is really a debug mode only feature. Does it even function in profile mode? I would suggest just disabling it in profile mode and commenting that it is debug mode only. That is probably less surprising than intentionally having a different default value in profile and debug mode.

Reply:
good point. done.